### PR TITLE
Add scrollbar to help window

### PR DESCRIPTION
**Pull request by Raphalex46**
This is my attempt at adding a scrollbar to the help window. It is heavily inspired by the code for the store sliders in `stores.cpp`.

I had to reduce the width of the text a bit to actually leave enough space for the scrollbar (also explicitly using this width value for the wrapping of the text).

Directly using the code from the stores UI code looks hard because it uses store-specific variables and the way the help window text and scroll works seems different as well.

This PR is for issue #5192

### CODE REVIEW

**Comment by AJenbo** [comment]
The code in stores.cpp it a big mess begging to get rewritten... glad you didn't reuse it :)

**Comment by Raphalex46** [comment]
Yeah, well what I did was kind of a mess as well then I think as I did it in the same style, using the information I had available about the help window.

However, the `help.cpp` code looks like a mess as well, it is already calling some functions from `stores.cpp`, so my changes are kind of building on the mess, which I don't like very much. Is the rewrite going to make it easier to re-use UI elements like this ?

File: Source/help.cpp [hunk 1]
```diff
@@ -138,6 +139,29 @@ int NumVisibleLines()
...
+		ClxDraw(out, { sliderXPos, sliderCurrent }, (*pSTextSlidCels)[13]);
+	}
+	ClxDraw(out, { sliderXPos, sliderCurrent }, (*pSTextSlidCels)[10]);
+	// Subtract visible lines from the total number of lines to get the actual
+	// scroll range
+	const float scrollRange = HelpTextLines.size() - NumVisibleLines();
+	// Subtract the size of the arrow buttons to get the length of the interior
+	// part of the slider
+	const float sliderLength = sliderCurrent - 12 - sliderStart;
+	ClxDraw(out, { sliderXPos, sliderStart + (int)(((float)SkipLines * sliderLength) / scrollRange) }, (*pSTextSlidCels)[12]);
```
Comment (AJenbo) [inline Source/help.cpp:162]: I'm not sure I see the point in having a float here, it's just subtracting and multiplying whole numberes. The cast back to int happens right after the division so the precision is discarded right away.

Comment (Raphalex46) [reply Source/help.cpp:162]: You're 100% right, I think I wanted to do it some other way at first and thought for some reason we needed the cast, but we really don't.

**Comment by AJenbo** [comment]
P.s. don't mind the failing ci/circleci: 3ds testflow, it's because a dependency ment 404 just a few minutes ago

File: Source/help.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/help.cpp:162]: Ok, if you change that to integer math then I'm happy to merge this :)

Comment (Raphalex46) [reply Source/help.cpp:162]: Should I add a commit to the PR ? Or is modifying the commit cleaner ? (I'm new to this as mentioned in the issue ahah)

**Comment by Raphalex46** [comment]
Yeah I figured this small change couldn't break the 3DS build :)

File: Source/help.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/help.cpp:162]: We prefer having things clean, but that also means that we will squash your PR in to one commit if it's messy so either is fine :)

Comment (AJenbo) [reply Source/help.cpp:162]:
```suggestion
	const int sliderLength = sliderCurrent - 12 - sliderStart;
	ClxDraw(out, { sliderXPos, sliderStart + (SkipLines * sliderLength / scrollRange) }, (*pSTextSlidCels)[12]);
```

Comment (Raphalex46) [reply Source/help.cpp]: Done :). Thank you for your quick answers !

**Comment by AJenbo** [comment]
Oh, I might have been a bit to quick with merging, could you look at this error on some of the targets:

```

/home/runner/work/devilutionX/devilutionX/Source/help.cpp:162:29: error: non-constant-expression cannot be narrowed from type 'unsigned int' to 'int' in initializer list [-Wc++11-narrowing]
        ClxDraw(out, { sliderXPos, sliderStart + ((SkipLines * sliderLength) / scrollRange) }, (*pSTextSlidCels)[12]);
```

**Comment by glebm** [comment]
The error was fixed in https://github.com/diasurgical/devilutionX/commit/4f21264a98ffb2fbd44b55ce887e0cca2234d274

**Pull request merged**